tests/operations.go

```diff
@@ -364,6 +364,70 @@ func (ops *Operations) IsPodRunningEventually(namespace, podName string) bool {
 		Should(BeTrue())
 }
 
+// GetnPods returns list of pod objects if list of pod count is matched to
+// expectedCount
+func (ops *Operations) GetnPods(namespace string, listOpts metav1.ListOptions, expectedCount int) (*corev1.PodList, error) {
+	podList, err := ops.PodClient.WithNamespace(namespace).
+		List(listOpts)
+	if err != nil {
+		return nil, errors.Wrapf(
+			err,
+			"failed to list pods with label {%s} in namespace {%s}",
+			listOpts.LabelSelector,
+			namespace,
+		)
+	}
+	if len(podList.Items) != expectedCount {
+		return nil, errors.Errorf(
+			"got more than one pod with label {%s} in namespace {%s}",
+			listOpts.LabelSelector,
+			namespace,
+		)
+	}
+	return podList, nil
+}
+
+// RestartPodEventually returns pod object after restarting the pod if it comes
+// to running state
+func (ops *Operations) RestartPodEventually(namespace string, listOpts metav1.ListOptions) (*corev1.Pod, error) {
+	podList, err := ops.GetnPods(namespace, listOpts, 1)
```

Comment:
let's take the `expectedCount` as argument to make generic and remove the count hardcodings

Reply:
```go
 RestartPodEventually(namespace string, expectedCount int , listOpts metav1.ListOptions) (*corev1.Pod, error) { }

```

Reply:
In pod restart scenarios we are mainly concerned about one pod, So hard coded to one. 
When it means generic we need to have below functions like:
```go
ArePodsRunningEventually(namespace string, listOpts metav1.ListOptions, expectedCount int) bool {
 // Logic to check  expectedCount pods are in running state or not
}
```
Similarly we need to have functions like 
```go
ArePodsDeletedEventually(namespace string, listOpts metav1.ListOptions, expectedCount int) bool {
// Logic
}
```
So should I make above changes?
cc: @prateekpandey14

Reply:
Updated func `RestartPodEventually` So If any one has to restart the m pods in n podList(m<n) we will have wrapper over `RestartPodEventually` function